packages/ui/src/memberships/components/AvatarInput.tsx

```diff
@@ -1,25 +1,35 @@
-import React from 'react'
+import React, { useCallback } from 'react'
 import { useFormContext } from 'react-hook-form'
 
 import { InputComponent, InputText } from '@/common/components/forms'
 import { SmallFileUpload } from '@/common/components/SmallFileUpload/SmallFileUpload'
 import { TextMedium } from '@/common/components/typography'
+import { resizeImageFile } from '@/common/helpers'
 
 export const AvatarInput = ({ initialPreview }: { initialPreview?: string }) => {
   const formContext = useFormContext()
 
+  const onUpload = useCallback(
+    async (event: React.ChangeEvent<HTMLInputElement>) => {
+      const file = event.target.files?.item(0) ?? null
+      if (file) {
+        try {
+          const fileBlob = await resizeImageFile(file, 192, 192, 'image/webp')
+          formContext.setValue('avatarUri', fileBlob, { shouldValidate: true })
+        } catch (e) {
+          formContext.setValue('avatarUri', file, { shouldValidate: true })
+        }
+      }
+    },
+    [formContext.setValue]
+  )
```

Comment:
Since the image is already resized inside SmallFileUpload why not just calling `onUpload(avatarPreview)` (maybe `avatarPreview` should be renamed to `localValue` and `localValue` to `file`).

So this would just be:
```suggestion
    (value) => formContext.setValue('avatarUri', value, { shouldValidate: true })
```

WDYT ?

Reply:
Right I forgot that `avatarPreview` was a url. But yes It would make more sense to me to resize the image just once and have the blob stored inside the `SmallFileUpload`.

Maybe `onUpload` could become:
```ts
(event: React.ChangeEvent<HTMLInputElement>, resizedImage?: Blob) => void
```